extension-manifest-v3/src/utils/engines.js

```diff
@@ -191,7 +203,7 @@ async function update(name) {
 
       const engineBytes = new Uint8Array(arrayBuffer);
       engine = FiltersEngine.deserialize(engineBytes);
-
+      engine.updateEnv(ENV);
```

Comment:
As we add some layers to the `FiltersEngine`, I would rather create a custom `deserialize` function, and put `updateEnv` call there. It would be then only in one place.